include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2477,47 +3160,1289 @@ namespace internal
-        3 * shape_info.dofs_per_component_on_face);
+          const unsigned int index =
+            direction == 0 ? subface_index % 2 : subface_index / 2;
+          return Eval_type(data.values_within_subface[index],
+                           data.gradients_within_subface[index],
+                           data.hessians_within_subface[index],
+                           data.fe_degree + 1,
+                           data.n_q_points_1d);
+        }
     }
 
-    /**
-     * Interpolate the values on the cell quadrature points onto a face.
-     */
-    template <bool do_evaluate, bool add_into_output>
     static void
-    interpolate_quadrature(
-      const unsigned int                            n_components,
-      const EvaluationFlags::EvaluationFlags        flags,
-      const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
-      const Number *                                input,
+    evaluate_in_face(const EvaluationFlags::EvaluationFlags evaluation_flag,
+                     const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
+                     Number *                                      values_dofs,
+                     Number *                                      values_quad,
+                     Number *           gradients_quad,
+                     Number *           hessians_quad,
+                     Number *           scratch_data,
+                     const unsigned int subface_index,
+                     const unsigned int face_no)
+    {
+      // TODO. Need both normal/tangent at 0 and 1 since it depends on the
+      // component!
+
+      Eval_normal eval_normal =
+        create_evaluator_tensor_product<Eval_normal>(shape_info.data.front(),
+                                                     subface_index,
+                                                     0);
+      Eval_tangent eval_tangent =
+        create_evaluator_tensor_product<Eval_tangent>(shape_info.data.back(),
+                                                      subface_index,
+                                                      1);
+
+      // Used for normal faces which are not anisotropic
+      Eval_general eval_general(shape_info.data.back().shape_values,
+                                shape_info.data.back().shape_gradients,
+                                shape_info.data.back().shape_hessians,
+                                shape_info.data.back().fe_degree + 1,
+                                shape_info.data.back().n_q_points_1d);
+
+      // Note, n_dofs on tangent face
+      const std::size_t n_dofs_tangent = shape_info.dofs_per_component_on_face;
+      const std::size_t n_dofs_normal =
+        n_dofs_tangent - Utilities::pow(fe_degree, dim - 2);
+
+      const std::size_t n_q_points =
+        fe_degree > -1 ?
+          Utilities::pow(n_q_points_1d, dim - 1) :
+          Utilities::pow(shape_info.data.front().n_q_points_1d, dim - 1);
+
+      const unsigned int face_orientation = face_no / 2;
```

Comment:
Can we choose a better name here, e.g., `face_direction`? The point is that the name 'orientation' is already used for rotations/mirroring of a face compared to the same face on a neighboring cell, so that would avoid some confusion here.